frontend/test/lit/test_decomposition.py

```diff
@@ -257,26 +257,26 @@ def test_decompose_singleexcitationplus():
     def decompose_singleexcitationplus(theta: float):
         # pylint: disable=line-too-long
         # CHECK-NOT: name = "SingleExcitationPlus"
-        # CHECK: [[a_scalar_tensor_float_2:%.+]] = stablehlo.constant dense<2.{{[0]+}}e+00>
-        # CHECK-NOT: name = "SingleExcitationPlus"
-        # CHECK: [[a_theta_div_2:%.+]] = stablehlo.divide %arg0, [[a_scalar_tensor_float_2]]
-        # CHECK-NOT: name = "SingleExcitationPlus"
-        # CHECK: [[b_scalar_tensor_float_2:%.+]] = stablehlo.constant dense<2.{{[0]+}}e+00>
-        # CHECK-NOT: name = "SingleExcitationPlus"
-        # CHECK: [[b_theta_div_2:%.+]] = stablehlo.divide %arg0, [[b_scalar_tensor_float_2]]
-        # CHECK-NOT: name = "SingleExcitationPlus"
         # CHECK: [[s0q1:%.+]] = "quantum.custom"({{%.+}}) {gate_name = "PauliX"
         # CHECK-NOT: name = "SingleExcitationPlus"
         # CHECK: [[s0q0:%.+]] = "quantum.custom"({{%.+}}) {gate_name = "PauliX"
         # CHECK-NOT: name = "SingleExcitationPlus"
+        # CHECK: [[a_scalar_tensor_float_2:%.+]] = stablehlo.constant dense<2.{{[0]+}}e+00>
+        # CHECK-NOT: name = "SingleExcitationPlus"
+        # CHECK: [[a_theta_div_2:%.+]] = stablehlo.divide %arg0, [[a_scalar_tensor_float_2]]
+        # CHECK-NOT: name = "SingleExcitationPlus"
         # CHECK: [[a_theta_div_2_scalar:%.+]] = "tensor.extract"([[a_theta_div_2]])
         # CHECK-NOT: name = "SingleExcitationPlus"
-        # CHECK: [[s1:%.+]]:2 = "quantum.custom"([[a_theta_div_2_scalar]], [[s0q0]], [[s0q1]]) {gate_name = "ControlledPhaseShift"
+        # CHECK: [[s1:%.+]]:2 = "quantum.custom"([[a_theta_div_2_scalar]], [[s0q1]], [[s0q0]]) {gate_name = "ControlledPhaseShift"
```

Comment:
Also curious about this change, the order of the qubits was swapped with no change in the example?

Reply:
That is the consequence of jax-sorting. The algorithm is deterministic, but the particular order of the equations might differ from test case to test case. Most of the tests just match the expectations but some of them don't. In these cases I needed to either change the algorithm or to change the test. Since I couldn't see clear rules to implement, I decided to change the test

Reply:
It is the consequence of jax-sorting. The sorting algorithm is deterministic, but the particular order of the equations might differ from test case to test case. Most of the tests just match the expectations but some of them don't. In these cases I needed to either change the algorithm or to change the test.

The point here is that we do not lit-check every line. The order of initial qubit-producing instructions must have been changed by the sorting (something we didn't capture in this test) so mlir assigned names in a different way.

Reply:
Got it, thanks :)

Reply:
I have added the `qinst` operation to the [forced_order_primitives](https://github.com/PennyLaneAI/catalyst/blob/2a9ed99f03af5e0a449c937632ae529651f51005/frontend/catalyst/utils/jax_extras.py#L163) set. Now the sort is much more restricted and this test has only two arithmetic operations reordered. Please check!

Reply:
Thanks 👍